### PR TITLE
Notifications rework

### PR DESCRIPTION
**Pull request by dpoirier**
* Make notification app more generally useful
* Create a generic 'send_notification()' internal API
* Everywhere we currently send mail, use the notification API
  instead of calling an email service more directly

### CODE REVIEW

**Comment by ntrncic** [comment]
please resolve conflicts

File: EquiTrack/utils/utils.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import json
```
Comment (ntrncic) [inline EquiTrack/utils/utils.py]: please move this to EquiTrack.utils.py

**Comment by domdinicola** [comment]
@dpoirier please update and resolv conflicts

File: EquiTrack/EquiTrack/factories.py [hunk 2]
```diff
@@ -77,32 +77,26 @@ class Meta:
     user = factory.SubFactory('EquiTrack.factories.UserFactory', profile=None)
 
 
+@factory.django.mute_signals(post_save)
```
Comment (domdinicola) [inline EquiTrack/EquiTrack/factories.py]: :+1:

**Comment by dpoirier** [comment]
Updated, conflicts resolved, tests passing again.

**Comment by dpoirier** [comment]
[ch4179]

**Comment by ntrncic** [comment]
please resolve conflicts

**Comment by ntrncic** [comment]
@dpoirier please resolve conflicts number 5. I hope it's the last one. sorry

**Comment by dpoirier** [comment]
@ntrncic conflicts have been resolved

**Pull request merged**